---
created: '2021-02-25T10:16:26.363162Z'
creator: sentry
source: tests/sentry/grouping/test_variants.py
---
app:
  hash: "418120a66f7031923031f5c52aca0724"
  component:
    app*
      exception*
        stacktrace*
          frame*
            function*
              "_main"
          frame (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start"
            package
              "std"
          frame (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start_internal"
            package
              "std"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::panicking::try::do_call"
            package
              "std"
          frame (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
            package
              "std"
          frame*
            function*
              "log_demo::main"
            package
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package
              "log"
          frame (non app frame)
            function*
              "sentry::integrations::log::Logger::log"
          frame (non app frame)
            function*
              "sentry::hub::Hub::with_active"
            package
              "sentry"
          frame (non app frame)
            function*
              "sentry::hub::Hub::with"
            package
              "sentry"
          frame (non app frame)
            function*
              "sentry::hub::Hub::with_active::{{closure}}"
            package
              "sentry"
          frame (non app frame)
            function (ignored unknown function)
              "<unknown>"
            package
              "sentry"
          frame (non app frame)
            function (ignored unknown function)
              "<redacted>"
            package
              "something"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
system:
  hash: "00719910980352c06ba93641057012e0"
  component:
    system*
      exception*
        stacktrace*
          frame*
            function*
              "_main"
          frame*
            function*
              "std::rt::lang_start"
            package
              "std"
          frame*
            function*
              "std::rt::lang_start_internal"
            package
              "std"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame*
            function*
              "std::panicking::try::do_call"
            package
              "std"
          frame*
            function*
              "std::rt::lang_start::{{closure}}"
            package
              "std"
          frame*
            function*
              "log_demo::main"
            package
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
          frame*
            function*
              "sentry::hub::Hub::with_active"
            package
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with"
            package
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with_active::{{closure}}"
            package
              "sentry"
          frame
            function (ignored unknown function)
              "<unknown>"
            package
              "sentry"
          frame
            function (ignored unknown function)
              "<redacted>"
            package
              "something"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
